test/new_relic/agent/stats_engine_test.rb

```diff
@@ -183,10 +183,14 @@ def test_record_scoped_and_unscoped_metrics_is_thread_safe
     nthreads = 25
     iterations = 100
 
-    nthreads.times do |tid|
-      threads << Thread.new do
-        iterations.times do
-          in_transaction('txn') do
+    # TODO: why are no scoped metrics being recorded?
```

Comment:
This is the test that fails if `instrumentation.thread.auto_instrument` is enabled

Reply:
First, I love that you added the specific configuration settings to each of these tests! 🎉 
Next, I wonder if there is a part of the StatsHash or Metric recording mechanisms that isn't getting a lock on it, but should be, causing the mismatched #'s?

Reply:
Actually, by moving the in_transaction block outside of the threads (since only one transaction can be happening at once, and now we can see inside threads) that solved that specific issue. Currently all the numbers match up consistently, however there are no scoped metrics being recorded, which is very odd. I haven't dug back into that issue yet, but that is where i left it previously. I suspect the inconsistent numbers we were seeing was due to the agent being unable to track more than one transaction at a time now that the threads are getting the agents state.